File: solutions/1630/1630-kir3i.cpp
class Solution {
public:
    unordered_set<int> s;
    int smax, smin;
    vector<bool> ans;
    bool ok;
    vector<bool> checkArithmeticSubarrays(vector<int>& nums, vector<int>& l, vector<int>& r) {
        int qsz = l.size();
        for (int q=0; q<qsz; q++) {
            s.clear();
            smax = -2e8;
            smin = 2e8;
            ok = true;
            for(int i=l[q]; i<=r[q]; i++) {
                s.insert(nums[i]);
                smax = max(smax, nums[i]);
                smin = min(smin, nums[i]);
            }
            
            int jump = (smax - smin) / (r[q] - l[q]);
            if (smin == smax) {
                ans.push_back(true);
                continue;
            } else if(!jump) {
                ans.push_back(false);
                continue;
            }
            for(int i=smin; i<=smax; i+=jump) {
                if(!s.count(i)) {
                    ans.push_back(false);
                    ok = false;
                    break;
                }
            }
            
            if (ok)
                ans.push_back(true);
        }
        return ans;
    }
};

File: solutions/1630/1630-kir3i.py
class Solution:
    def checkArithmeticSubarrays(self, nums, l, r):
        return [self.isArithmetic(nums[lo:hi+1]) for lo, hi in zip(l, r)]

    def isArithmetic(self, nums):
        maxn = max(nums)
        minn = min(nums)
        if maxn == minn:
            return True

        jump = (maxn - minn) // (len(nums) - 1)
        if jump == 0:
            return False
        setNums = set(nums)
        for x in range(minn, maxn + 1, jump):
            if x not in setNums:
                return False
        return True